watchdog/src/metrics.rs

```diff
@@ -75,19 +101,45 @@ fn encode_metrics(w: &mut MetricsEncoder<Vec<u8>>) -> std::io::Result<()> {
         .value(&[("code", "ahead")], ahead)?
         .value(&[("code", "behind")], behind)?;
 
+    let api_access = crate::storage::get_api_access();
+    w.encode_gauge(
+        "api_access_target",
+        flag_to_gauge(api_access.target),
+        "Expected value of the Bitcoin canister API access flag.",
+    )?;
+    w.encode_gauge(
+        "api_access_actual",
+        flag_to_gauge(api_access.actual),
+        "Actual value of the Bitcoin canister API access flag.",
+    )?;
+    w.encode_gauge(
+        "api_access_is_in_sync",
+        if api_access.is_in_sync() { 1.0 } else { 0.0 },
+        "Whether the actual and target values of the Bitcoin canister API access flag are in sync.",
+    )?;
+
     let mut available_explorers = HashMap::new();
     for explorer in health.explorers {
         available_explorers.insert(explorer.provider.clone(), explorer);
     }
     let mut gauge = w.gauge_vec("explorer_height", "Heights from the explorers.")?;
-    for explorer in BitcoinBlockApi::network_explorers(bitcoin_network) {
+    let mut available_explorers_count: u64 = 0;
+    for explorer in BitcoinBlockApi::network_explorers(config.bitcoin_network) {
         let height = available_explorers
             .get(&explorer)
             .map_or(NO_VALUE, |block_info| {
                 block_info.height.map_or(NO_VALUE, |x| x as f64)
             });
+        if !height.is_nan() {
+            available_explorers_count += 1;
+        }
         gauge = gauge.value(&[("explorer", &explorer.to_string())], height)?;
     }
+    w.encode_gauge(
+        "available_explorers",
+        available_explorers_count as f64,
+        "The number of available explorers to compare against.",
+    )?;
```

Comment:
If I understand this correctly, an explorer would be marked as available as long as it responded at least once in the past, but if it has, say, been offline or unresponsive for several days, it would still be considered available by this logic, correct? Is that the intent? Would we still want to consider these explorers available in this case?

Reply:
No, it's not exactly right. Currently the information from explorers is stored till the next fetch, so if explorer failed to provide data it will be erased. That's why the number of explorers can vary from fetch to fetch (every 2 minutes currently), and that's why we have `min_explorers` threshold of 2 with 6 explorers polled.